pkgs/development/python-modules/huawei-solar/default.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  fetchFromGitLab,
+}:
+let
+  python = python3.override {
+    self = python;
+    packageOverrides = (
+      self: super: {
+        pymodbus = super.pymodbus.overridePythonAttrs (oldAttrs: rec {
+          version = "3.6.9";
+          src = fetchFromGitHub {
+            inherit (oldAttrs.src) owner repo;
+            rev = "refs/tags/v${version}";
+            hash = "sha256-ScqxDO0hif8p3C6+vvm7FgSEQjCXBwUPOc7Y/3OfkoI=";
+          };
+        });
+      }
+    );
+  };
+
```

Comment:
Package overrides are not supported in the python package set as that will introduce multiple versions of a package into the PYTHONPATH that will only ever use the one that appears first.

That being said, the home-assistant python package set also pins 3.6.9, so you can safely drop this override and set `meta.broken = lib.versionAtLeast pymodbus "3.7.0"`.

Reply:
That is correct. Because python3Packages has a newer pymodbus version as you discovered. Applications can manage package overrides, since they're at the leaf. Nothing is going to conflict with their override choices. Libraries cannot.